### DISCUSSION TITLE
DeviceScript board for ESP32-S3

### DISCUSSION THREAD

**Discussion by jus101** [microsoft/devicescript#487] [Q&A] [answered]
I bought a few ESP32-S3 dev boards recently to specifically to tinker with DeviceScript.. except I confused S2/C3/C2/S3 (damn dyslexia).

I've had a quick look through the documentation, tried to create a bare board JSON definition based off the ESP32-S2, run build, and flashed the firmware, but all I get is ```invalid header: 0xffffffff``` from the serial on repeat... so I'm guessing it's more complicated than this.

What is involved in getting another board like the S3 working with DeviceScript? Is it a big thing, or something minor that I might be able to tackle? Any pointers on where I could start?

**Comment by mmoskal** [comment]
It shouldn't be too difficult if you are familiar with `esp-idf`. I may take a look at this in the next few months but no promises. PRs are always welcome!

**Reply by pelikhan** [reply]
@jus101 please try again. we added a definition for the https://microsoft.github.io/devicescript/devices/esp32/esp32s3-devkit-m . Which dev board did you get?

**Comment by pelikhan** [comment, marked as answer]
The support for esp32-s3 is now documented at https://microsoft.github.io/devicescript/devices#implementation-status
